weather.py

```diff
@@ -1,72 +1,143 @@
-    # it in as a flag, or reading from the environment.
-    apikey = 'bTWRzyxjP94K3Wn66NlQsr3h2qppp3cs'
-
-    # Define the api path for getting locations.
-    locations_path = 'locations/v1/regions'
-
-    # Here we call the get_locations() function passing in the locations url
-    # and api key.
-    locations = get_locations('{}/{}?apikey={}'.format(
-        api_endpoint, locations_path, apikey))
-
-    # ...and for example, let's say we want to know Africa's ID...
-    print(locations['Africa'])
-    # or Europe for that matter...
-    print(locations['Europe'])
-    # ...and let's pretend the user specifed the value as a command-line
-    # argument...
-    location = 'Central America'
-    print(locations[location])
-    # worth noting that what if the user specifed the location in lower-case?
-    # we could add all the locations to the dict in lower-case, then convert
-    # the user specifed location to lower-case too.
-    # Starting to sound like how to store stuff in a database. No coincidence!
+    apikey = '4bAzYrckrKgStAXpCiekZLyvkAGkHLmG'
+
+    # If user uses lower case,  we could add all the locations to the dict in lower-case, 
+    # then convert
+    
+    # Location to get the weather for
+    city = 'York'
+    # Admin area
+    admin_areas_path = 'locations/v1/adminareas'
+
+    admin_areas = get_locations('{}/{}?apikey={}'.format(
+        api_endpoint, admin_areas_path, apikey))
+
+    admin_area_code = (admin_areas[(city)])
+
+    # City
+    city_path = 'locations/v1/cities/search'
+    
+    search = '{} {}'.format(
+            city, admin_area_code)
+
+    url = '{}/{}?apikey={}&q={}'.format(
+        api_endpoint, city_path,  apikey, search )
+    
+    locations = make_request(url)
+
+    location: Location = dacite.from_dict(Location,locations)
+    location_key = location.Key
+
+    # Conditions
+    conditions_path = 'currentconditions/v1'
+
+    url = '{}/{}/{}?apikey={}'.format(
+        api_endpoint, conditions_path, location_key, apikey)
+
+    out = make_request(url)
+    conditions: Weather = dacite.from_dict(Weather,out)
+
+    print("The Current Conditions for " + location.LocalizedName +  " are:")
+    date_time = conditions.LocalObservationDateTime
+    date_time_list = date_time.split("T")
+    date = (date_time_list[0])
+    print("Date: " + str(date))
+    time_long = (date_time_list[1])
+    time_list = time_long.split(":")
+    time_hrs = (time_list[0])
+    time_mins = (time_list[1])
+    print("Time: " + str(time_hrs) + ":" + str(time_mins) + "hrs")
+    print("Summary: " + conditions.WeatherText)
+    value = conditions.Temperature.Metric.Value
+    unit = conditions.Temperature.Metric.Unit
+    temp = str(value) + unit
+    print("Temperature: " + temp)
+    rain =  conditions.HasPrecipitation
+    print("Precipitation: " + str(rain))
+    daylight = conditions.IsDayTime
+    if(daylight == True):
+        print("Its daytime")
+    elif(daylight == False):
+        print("Its night")
```

Comment:
🤮  Add some newlines the break up the code.
Maybe group related steps together, so all the `date` related stuff is together, then a newline, then al;; the `time` related stuff etc etc.

```
if daylight:
   print("It's daytime")
   return

print("It's night")
return
```

As `if ` evaluates an expression as true or false, simply testing `daylight` will return true or false.
If it's `True` we print as return from the `run()` function.
Otherwise we just continue and print the other statement and then return

Much much nicer.

Reply:
I thought there must be a better way of doing this. I've updated as suggested. I did have a look at 'dateutil' in order to print the date in a sentence.